nixos/modules/config/stevenblack.nix

```diff
@@ -0,0 +1,35 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.networking.stevenblack;
+
+  activatedHosts = with cfg.block; [ ]
+    ++ optionals fakeNews [ "fakenews" ]
+    ++ optionals gambling [ "gambling" ]
+    ++ optionals porn [ "porn" ]
+    ++ optionals social [ "social" ];
+
+  hostsPath = "${pkgs.stevenblack-blocklist}/alternates/" + concatStringsSep "-" activatedHosts + "/hosts";
+in
+{
+  options.networking.stevenblack = {
+    enable = mkEnableOption "Enable the stevenblack hosts file blocklist.";
+
+    block = {
```

Comment:
Would be better as a list of enums iirc it's called

Reply:
Hmm, I don't think it's possible to do it that way, the blocklist allows for multiple combinations and they have to be in a specific order (i.e. if `fakenews` is activated then it has to be the first in the `activatedHosts` list so the folder exists [here](https://github.com/StevenBlack/hosts/tree/master/alternates)).

Reply:
i rebased the pr and switched to use elem

Reply:
looks good